src/rgw/rgw_op.h

```diff
@@ -1863,6 +1863,10 @@ class RGWCompleteMultipart : public RGWOp {
   bufferlist data;
   std::unique_ptr<rgw::sal::MPSerializer> serializer;
   jspan_ptr multipart_trace;
+  std::unique_ptr<rgw::sal::MultipartUpload> upload;
```

Comment:
i don't think that you need `upload` and `target_obj` as data members.
from `upload` we need the time stamp and from `target_obj` the version id